src/compiler/c.rs

```diff
@@ -286,7 +286,7 @@ where
             .ok()
             .map(|f| {
                 f.flatten()
-                    .filter(|f| f.path().extension().map_or(false, |ext| ext == "bc"))
+                    .filter(|f| f.path().extension().is_some_and(|ext| ext == "bc"))
```

Comment:
What about migrating OpenSSL-unrelated changes to the dedicated PR?
I don't think that changes breaks CI, but this split will minify change set, that is related to failure.

Reply:
Sure